building/buildah_speed/buildah_speed.sh

```diff
@@ -0,0 +1,252 @@
+
+The container can then use the images without having to pull them into it's own
+local storage.  This gives you excellent security, with the same performance as
+buildah_fast demonstration.
+
+Build a container image with shared read/only containers/storage.
+"
+    echo
+    sudo rm -rf $MEDIUM
+    sudo mkdir -p $MEDIUM
+    sudo chcon -t container_file_t $MEDIUM
+    read_color "    sudo podman run --device /dev/fuse -ti -v $PWD/Dockerfile:/Dockerfile:Z -v /var/lib/containers/storage:/var/lib/shared:ro -v $MEDIUM:/var/lib/containers quay.io/buildah/stable buildah bud /"
+    sudo /bin/time podman -o /tmp/buildah_medium_bud.txt --format %e run --device /dev/fuse -ti -v $PWD/Dockerfile:/Dockerfile:Z -v /var/lib/containers/storage:/var/lib/shared:ro -v $MEDIUM:/var/lib/containers quay.io/buildah/stable buildah bud /
+
+    echo_color "
+    Completed in $(cat /tmp/buildah_medium_bud.txt) seconds
+"
+    echo
+    read -p "Enter to continue"
+    clear
+}
+
+buildah_medium_bud_with_overlay() {
+    echo
+    echo_color "
+In this example, we are going to mount in the containers/storage read/only from
+the host into the containers as an additional store at /var/lib/shared. We are
+also using an OverlayMount of /var/cache/dnf to help speed up the dnf deployment
+inside of the container.
+
+Build a container image with shared read/only containers/storage with overlay mount.
+"
+    echo
+    sudo rm -rf $MEDIUM
+    sudo mkdir -p $MEDIUM
+    sudo chcon -t container_file_t $MEDIUM
+    read_color "    sudo podman run --device /dev/fuse -ti -v $PWD/Dockerfile:/Dockerfile:Z -v /var/lib/containers/storage:/var/lib/shared:ro -v $MEDIUM:/var/lib/containers quay.io/buildah/upstream buildah bud /"
+    sudo /bin/time -o /tmp/buildah_medium_bud_with_overlay.txt --format "%e" podman run --device /dev/fuse -ti -v $PWD/Dockerfile:/Dockerfile:Z -v /var/lib/containers/storage:/var/lib/shared:ro -v $MEDIUM:/var/lib/containers quay.io/buildah/upstream buildah bud /
+
+    echo_color "
+    Completed in $(cat /tmp/buildah_medium_bud_with_overlay.txt) seconds
+"
+    echo
+    read -p "Enter to continue"
+    clear
+}
+
+speed_table() {
+    echo "
+    As you can see below, the slowest buildah pull to multiple times as long
+    as the fastest, while the medium took relatively the same amount of time
+    as the fastest.
+
+    "
+    printf "    _______________________________________\n"
+    printf "    | %-10s | %-10s| %-10s |\n" "Slowest" "Medium" "Fastest"
+    printf "    | ---------- | --------- | ---------- |\n"
+    printf "    | %-10s | %-10s| %-10s |\n" $(cat /tmp/buildah_slowest.txt) $(cat /tmp/buildah_medium.txt) $(cat /tmp/buildah_fastest.txt) 
+    printf "    |_____________________________________|\n"
+    echo
+    echo
+    read -p "Enter to continue"
+    clear
+}
+
+clean_temp_files() {
+    sudo rm -rf /tmp/buildah*txt $SLOWEST $MEDIUM
+}
+
+clear
+
+buildah_medium_bud
+
+#setup
```

Comment:
This shouldn't be commented out should it?

Reply:
oops, debugging code.  Fixed